### DISCUSSION TITLE
Kubernetes-client docs / token auto refresh

### DISCUSSION THREAD

**Discussion by kobaltoxid** [fabric8io/kubernetes-client#4618] [Q&A]
Hello,

I am writing to you since I am interested in implementing the new feature available as of version 6.1.0 which would be the automatic token refresh for AWS clusters. I have tried checking your docs if you have an example or some information on the topic, but I can't seem to find anything specific. Would you mind pointing me in the right direction?

I am currently running version 6.2.0 and I'm using `new KubernetesClientBuilder().withConfig(config).build()` keep in mind that my Config object is built upon `Config config = fromKubeconfig(kubeconfigFile)`.

The current result I have, is that after a while, the cluster would lose authorization. The error thrown is from `          config.getErrorMessages().put(401, "Unauthorized! Token may have expired! Please log-in again.");` on line 709 in Config.java

Thank you for sparing the time!

**Comment by manusa** [comment]
You can take a look to the [`TokenRefreshInterceptor`](https://github.com/fabric8io/kubernetes-client/blob/e3ae1c675d2d40d7e34a3c692d8a05711f1ea817/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/TokenRefreshInterceptor.java#L33) which does this for  OIDC tokens.
And the [`Config.refresh`](https://github.com/fabric8io/kubernetes-client/blob/e3ae1c675d2d40d7e34a3c692d8a05711f1ea817/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/Config.java#L613) method (called by the interceptor) which re-reads the configuration (kube config, service account).
